src/Streamly/Internal/Data/Stream/IsStream/Nesting.hs

```diff
@@ -1751,6 +1751,7 @@ data SessionState t m k a b = SessionState
     , sessionOutputStream :: t (m :: Type -> Type) (k, b) -- ^ Completed sessions
     }
 
+-- XXX Need to check where to cleanup
```

Comment:
This is still incomplete. I Will get back to this or the reviewer can help me out if the info is known.